src/redux/selectors.ts

```diff
@@ -57,11 +59,138 @@ export const isInPreviewModeSelector = createSelector(
 
 export const isInClusterModeSelector = createSelector(
   (state: RootState) => state,
-  appState =>
-    Boolean(appState.main.previewResourceId && appState.main.previewResourceId.endsWith(appState.config.kubeconfigPath))
+  ({main, config}) => {
+    const kubeConfigPath = config.projectConfig?.kubeConfig?.path || config.kubeConfig.path;
+    if (kubeConfigPath) {
+      return Boolean(main.previewResourceId && main.previewResourceId.endsWith(kubeConfigPath));
+    }
+    return false;
+  }
 );
 
 export const logsSelector = createSelector(
   (state: RootState) => state.logs.logs,
   logs => logs.join('\n')
 );
+
+export const activeProjectSelector = createSelector(
+  (state: RootState) => state.config,
+  config => config.projects.find(p => p.rootFolder === config.selectedProjectRootFolder)
+);
+
+export const currentConfigSelector = createSelector(
+  (state: RootState) => state.config,
+  config => {
+    const applicationConfig: ProjectConfig = populateProjectConfig(config);
+    const projectConfig: ProjectConfig | null | undefined = config.projectConfig;
+    return mergeConfigs(applicationConfig, projectConfig);
+  }
+);
+
+// Needs better way to do it!!
+export const mergeConfigs = (baseConfig: ProjectConfig, config?: ProjectConfig | null) => {
+  if (!(baseConfig && baseConfig.settings && baseConfig.kubeConfig)) {
+    throw Error('Base config must be set');
+  }
+
+  if (!config) {
+    return baseConfig;
+  }
+
+  if (
```

Comment:
Adding new options here will add new code and new complexity. Do you think it is a good idea to create config where we map each key and a handler or something similar that we should do. All boolean/string... specific types seem the same logic with different keys